esphome/components/ags10/sensor.py

```diff
@@ -0,0 +1,122 @@
+import esphome.codegen as cg
+from esphome import automation
+import esphome.config_validation as cv
+from esphome.components import i2c, sensor
+from esphome.const import (
+    CONF_ID,
+    ICON_RADIATOR,
+    ICON_RESTART,
+    DEVICE_CLASS_VOLATILE_ORGANIC_COMPOUNDS_PARTS,
+    STATE_CLASS_MEASUREMENT,
+    UNIT_PARTS_PER_BILLION,
+    CONF_ADDRESS,
+    CONF_TVOC,
+    CONF_VERSION,
+    CONF_MODE,
+    CONF_VALUE,
+)
+
+CONF_RESISTANCE = "resistance"
+
+DEPENDENCIES = ["i2c"]
+
+ags10_ns = cg.esphome_ns.namespace("ags10")
+AGS10Component = ags10_ns.class_("AGS10Component", cg.PollingComponent, i2c.I2CDevice)
+
+# Actions
+AGS10NewI2cAddressAction = ags10_ns.class_(
+    "AGS10NewI2cAddressAction", automation.Action
+)
+AGS10SetZeroPointAction = ags10_ns.class_("AGS10SetZeroPointAction", automation.Action)
+
+CONFIG_SCHEMA = (
+    cv.Schema(
+        {
+            cv.GenerateID(): cv.declare_id(AGS10Component),
+            cv.Required(CONF_TVOC): sensor.sensor_schema(
```

Comment:
All sensors are to be Optional
```suggestion
            cv.Optional(CONF_TVOC): sensor.sensor_schema(
```

Reply:
Done